### PR TITLE
add pipline design pattern project

### PR DESCRIPTION
**Pull request by mojtabavali**

### CODE REVIEW

**Comment by mohammadKarimi** [comment]
there are some mistakes which you can fix them as soon as passible before merging:

1- The ExceptionHandling method catches all exceptions and prints a generic "catched" message with the exception message.
2- You can use  specific exception types such as CountryBlockedException instead throw new Exception.
3- You can use const pattern matching instead of == operator.
4- I prefer to add required keyword to IP property. because i want to show the framework this property is required on construction phase.
5- finally, Delegate Naming. for practice is good. but in real application you must use Action instead of creating a new delegate type.

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]
Please follow my comments.

**Comment by mojtabavali** [comment]
thanks for your comment. I changed codes base on your comment.

File: src/MojtabaValizadeh/PipelineDesignPattern/CountryBlockedException.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿namespace PipelineDesignPattern;
+
+public class CountryBlockedExceptionHandler:Exception
+{
+    public CountryBlockedExceptionHandler() : base() { }
+
+    public CountryBlockedExceptionHandler(string message) : base(message) { }
```
Comment (mohammadKarimi) [inline src/MojtabaValizadeh/PipelineDesignPattern/CountryBlockedException.cs:7]: Perfect 👍

File: src/MojtabaValizadeh/PipelineDesignPattern/Framework.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using Dumpify;
+
+namespace PipelineDesignPattern;
+
+public class Framework
+{
+    public void Authentication(Action<HttpContext> func,HttpContext httpContent)
+    {
+        if(httpContent.Ip is "192.168.0.130")
```
Comment (mohammadKarimi) [inline src/MojtabaValizadeh/PipelineDesignPattern/Framework.cs:9]: perfect.

Comment (mohammadKarimi) [reply src/MojtabaValizadeh/PipelineDesignPattern/Framework.cs:9]: however you consider that you should check the nullability of httpContext before.

File: src/MojtabaValizadeh/PipelineDesignPattern/Framework.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        func(httpContent);
+    }
+
+    public void ExceptionHandling(Action<HttpContext> func, HttpContext httpContent)
+    {
+        try
+        {
+            func(httpContent);
+        }
+        catch (CountryBlockedExceptionHandler e)
```
Comment (mohammadKarimi) [inline src/MojtabaValizadeh/PipelineDesignPattern/Framework.cs:22]: perfect.

**Review by mohammadKarimi — APPROVED** [review]
Perfect.

**Pull request merged**